src/core/lib/gprpp/time.h

```diff
@@ -200,6 +200,16 @@ class Duration {
     }
     return *this;
   }
+  Duration& operator*=(int64_t multiplier) {
+    if (millis_ == std::numeric_limits<int64_t>::max()) {
+      *this = multiplier < 0 ? NegativeInfinity() : Infinity();
+    } else if (millis_ == std::numeric_limits<int64_t>::min()) {
+      *this = multiplier < 0 ? Infinity() : NegativeInfinity();
+    } else {
+      millis_ *= multiplier;
```

Comment:
Need to deal with overflow here...

either explicitly (I probably made a mistake):
```
} else if (multiplier > 0 && millis_ > std::numeric_limits<int64_t>::max() / multiplier) {
  *this = Infinity();
} else if (multiplier < 0 && millis_ > std::numeric_limits<int64_t>::max() / -multiplier) {
  *this = NegativeInfinity();
} else if (multiplier > 0 && millis_ < std::numeric_limits<int64_t>::min() / multiplier) {
  *this = NegativeInfinity();
} else if (multiplier < 0 && millis_ < std::numeric_limits<int64_t>::min() / -multiplier) {
  *this = Infinity();
}
```

or leveraging the code for double and replacing this method with:
```
Duration& operator*=(int64_t multiplier) { *this = *this * static_cast<double>(multiplier); }
```

Reply:
Good catch, fixed.